lm_eval/evaluator.py

```diff
@@ -51,23 +52,40 @@ def generate_text(self, task_name):
             n_tasks=n_tasks,
             args=self.args,
         )
-        references = [task.get_reference(dataset[i]) for i in range(n_tasks)]
+        if self.args.load_references_path:
```

Comment:
I made references optional in this branch because I explored different references. If you find this extra flexibility redundant, please feel free to remove it.